pandas/tests/reshape/merge/test_multi.py

```diff
@@ -481,6 +481,56 @@ def test_merge_datetime_index(self, klass):
         result = df.merge(df, on=[df.index.year], how="inner")
         tm.assert_frame_equal(result, expected)
 
+    def test_merge_datetime_multi_index_empty_df(self):
+
+        midx1 = pd.MultiIndex.from_tuples(
+            [[pd.Timestamp("1950-01-01"), "A"], [pd.Timestamp("1950-01-02"), "B"]],
+            names=["date", "panel"],
+        )
+        left = DataFrame(
+            data={
+                "data": [1.5, 1.5],
+            },
+            index=midx1,
+        )
+
+        midx2 = pd.MultiIndex.from_tuples([], names=["date", "panel"])
+
+        right = DataFrame(index=midx2, columns=["state"])
+
+        midx3 = pd.MultiIndex.from_tuples(
+            [[pd.Timestamp("1950-01-01"), "A"], [pd.Timestamp("1950-01-02"), "B"]],
+            names=["date", "panel"],
+        )
+
+        expected_left_merge = DataFrame(
+            data={
+                "data": [1.5, 1.5],
+                "state": [None, None],
+            },
+            index=midx3,
+        )
+
+        expected_right_merge = DataFrame(
+            data={
+                "state": [None, None],
+                "data": [1.5, 1.5],
+            },
+            index=midx3,
+        )
+
+        result_left_merge = left.merge(right, how="left", on=["date", "panel"])
```

Comment:
can you paramaterize this; it is very hard to understand what is equal to what here. you *can* put the expected things in the parametr. I would like to hard code the result data frame (rather than trying to do operations to get it). its *very* hard to read otherwise.

Reply:
I parametrized the merge type in my commit -- did you want me to explicitly construct an expected data frame in the parametrize call here? https://github.com/pandas-dev/pandas/pull/36897/files#diff-513c054eb6b1b4b5767980db8c8dc0bd93f7ea55df9facea3465bfac11785474R509
That feels like it might be difficult to read but I can do it.